lib/requester/util.js

```diff
@@ -67,18 +71,40 @@ var _ = require('lodash'),
      *
      * @param lookupOptions
      * @param lookupOptions.port
-     * @param lookupOptions.restrictedAddresses
+     * @param lookupOptions.network
+     * @param lookupOptions.network.restrictedAddresses
+     * @param lookupOptions.network.hosts
+     * @param lookupOptions.network.hosts.type
      * @param hostname
      * @param options
      * @param callback
      */
     lookup = function (lookupOptions, hostname, options, callback) {
-        var self = this;
+        var self = this,
+            lowercaseHost = hostname.toLowerCase(),
+            networkOpts = lookupOptions.network || {},
+            hostsOptions = networkOpts.hosts,
+            hostIpMap,
+            resolvedFamily = 4,
+            resolvedAddr;
+
+        hostsOptions && hostsOptions.type === HOSTS_TYPE.HOST_IP_MAP &&
+            (hostIpMap = hostsOptions[HOSTS_TYPE.HOST_IP_MAP]) && (resolvedAddr = hostIpMap[lowercaseHost]);
+
+        if (resolvedAddr) {
+            _.includes(resolvedAddr, COLON) && (resolvedFamily = 6);
+
+            // returning error synchronously causes uncaught error
+            return setImmediate(function () {
+                callback(self.isAddressRestricted(resolvedAddr, networkOpts) ?
```

Comment:
Could you extract the lookup function so that `dns.lookup` and `hostIpMap` lookup are passed with the same callback and `isAddressRestricted` is handled in there?

This looks prone to errors.